### DISCUSSION TITLE
Data Explorer mode for lists (of lists)

### DISCUSSION THREAD

**Discussion by artmg** [posit-dev/positron#4717] [Q&A] [answered]
One of my key development workflows in RStudio is to access data from interfaces that return nested lists (JSON style), and to explore the structure and linkages within that hierarchical data to be able to then convert it into meaningful tabular datasets for further analysis and processing. It is useful to be able to open and close nodes to reveal or hide the subelements, to prevent the overall length (if flattened) from overwhelming one's ability to make sense of the structure and content.

The Positron Data Explorer does not currently seem to have a mode for browsing lists, lists of lists and more deeply nested structures. Does such a feature appear on your roadmap, yet?

**Comment by petetronic** [comment, marked as answer]
Hi @artmg, thank you for trying out Positron! We've discussed the idea of an object viewer, as well as better list views, given the variables pane is not really intended for deep exploration. We have this issue that is related https://github.com/posit-dev/positron/issues/3028 if you'd like to weigh in there?

**Reply by artmg** [reply]
Thanks Pete I have. So the summary is that its currently on the 'unscheduled' pile of the roadmap, but at least users can do some kind of object browsing already in the space-limited Variables pane. 

I guess that if they were that desperate, users could install the package [listviewer](https://cran.r-project.org/package=listviewer) and use interactive html-based `listviewer::jsonedit` whilst waiting for that issue to bubble up in importance.
